### ISSUE TITLE
Command history

### ISSUE THREAD

**Issue by massimocode** [ow--/vscode-subword-navigation#9]
Hey there

So I've been using this for several months now and it's absolutely awesome.

However there is one thing that is really annoying.

If you navigate within code VS Code's normal word navigation or arrow keys (i.e. not the subword navigation), then press undo, it undoes your actual last code change (as opposed to the navigation)

But if you navigate using subword then it undoes the navigation. This means the undo command does not work consistently with VS Code's normal navigation.

Is there any chance this can be fixed? I wanted to do it as a PR but a quick look at the VS Code documentation makes no mention of history or undo.

**Comment by massimocode** [comment]
Hey I managed to fix this and put in a PR
https://github.com/ow--/vscode-subword-navigation/pull/10

It seems when you register an editor command it automatically stores the history of the editor before the command took place. But we don't really want that because that will store the cursor position when navigating and so undo will literally undo our navigation (not something we want). So I am registering the commands as normal commands as opposed to editor commands.

Another issue I found is that when we are deleting stuff, the editor internally registers its own undo in the history as well. So you have to undo twice to be able to completely undo a subword delete (although doing it once is enough to undo the code change, you need to do it again to get rid of the duff history point that the wrapped command put in). So whilst when we try to undo 1 subword delete, we only need to press undo once, the interesting thing is that when we try to undo 2 consecutive subword deletes, then we have to press undo 3 times, and for 3 consecutive subword deletes we would have to press undo 5 times, and so on... So I registered these commands as normal ones too and the undo functionality now works as expected.

Let me know what you think!

**Comment by ow--** [comment]
Stupid tired head, the PR comment would've made much more sense here.

Thanks!

**Comment by gordey4doronin** [comment]
I've been using the plugin for one week and also noticed this strange history behavior.
It's cool it has been fixed already.
@massimocode Thanks!

@ow-- I checked the version of the plugin in vscode marketplace is `1.1.3`.
However, there are no tags in this repository.
So it's not clear at first glance which version exactly is published to the marketplace.
Of course looking in the commits makes sense, and I see the commit with comment saying **version bump**.

Don't you want to mark it f286736130772c27a86b592ad59355b01749cbc2 with `v1.1.3` tag?
